src/Servers/IIS/AspNetCoreModuleV2/InProcessRequestHandler/inprocessapplication.cpp

```diff
@@ -545,8 +549,10 @@ void
 IN_PROCESS_APPLICATION::HandleRequestCompletion()
 {
     SRWSharedLock lock(m_stateLock);
-    auto requestCount = m_requestCount--;
-    if (m_fStopCalled && requestCount == 0)
+    m_requestCount--;
```

Comment:
Why did you separate this statement? You can end up calling callback multiple times.